osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/SliderBodyPiece.cs

```diff
@@ -26,6 +26,10 @@ public SliderBodyPiece()
             {
                 AccentColour = Color4.Transparent
             };
+
+            // SliderSelectionBlueprint relies on calling ReceivePositionalInputAt on this drawable to determine whether selection should occur.
+            // Without AlwaysPresent, a movement in a parent container (ie. the editor composer area resizing) could cause incorrect input handling.
+            AlwaysPresent = true;
```

Comment:
Nit, but in my opinion, I think it's a better choice to set `AlwaysPresent` on the place where the component is set to be hidden, rather than inside the component class itself, so in this case this will be put in `SliderSelectionBlueprint` in the same place where `AccentColour = Color4.Transparent` is set for the body piece:

![CleanShot 2021-05-06 at 22 27 15](https://user-images.githubusercontent.com/22781491/117354627-39b14100-aeba-11eb-9ebd-3313b6cc43d3.png)

Reply:
...what? as far as I can see the *parent* is not present.

![2021-05-06-214416_838x855_scrot](https://user-images.githubusercontent.com/20418176/117356774-6bbfa480-aeb4-11eb-9a3d-eb7feb81becc.png)

Reply:
ah, nevermind me, I've been confused by what's causing the slider body piece to not be present, though I still don't understand why it's not present, it seems to be hidden by having `Alpha = 0`, but I can't find where that's done.

Reply:
i couldn't find it either, so i jerry-rigged a fake conditional breakpoint with a framework checkout. the call stack responsible for the hide is:

```
SliderSelectionBlueprint.LoadComplete();
SelectionBlueprint<HitObject>.LoadComplete();
SelectionBlueprint<HitObject>.updateState();
SliderSelectionBlueprint.OnDeselected();
SelectionBlueprint<HitObject>.OnDeselected();
```

however this call stack being this generic makes me wonder whether other blueprints that delegate to their drawable contents in `ReceivePositionalInputAt()` are affected as well. it sure as hell feels like they should be.

Reply:
I've rewritten this to avoid the reference to non-visible pieces (generally how other blueprints are doing things already). See is that reads better.

Reply:
That does read better and sound like a good solution 👍🏼